.ci/buildbot/worker.py

```diff
@@ -0,0 +1,471 @@
+    variable BUILDBOT_CLEAN when in the GUI the option "Clean source code and
+    build directory" is checked by the user. The 'clean_obj' build property and
+    the BUILDBOT_CLEAN_OBJ environment variable will be set when either the
+    "Clean build directory" GUI option is set, or the master detects a change
+    to a CMakeLists.txt or *.cmake file.
+
+    Parameters
+    ----------
+    scriptpath
+        Pass __file__ from the main builder script.
+    llvmsrcroot
+        Absolute path to the llvm-project source checkout. Since the builder
+        script is supposed to be a part of llvm-project itself, the builder
+        script can compute it from __file__.
+    parser
+        Use this argparse.ArgumentParser instead of creating a new one. Allows
+        adding additional command line switched in addition to the pre-defined
+        ones. Build script are encouraged to apply the pre-defined switches.
+    clobberpaths
+        Directories relative to workdir that need to be deleted if the build
+        configuration changes (due to changes of CMakeLists.txt or changes of
+        configuration parameters). Typically, only source checkouts are not
+        deleted.
+    workerjobs
+        Default number of build and test jobs; If set, expected to be the number
+        of jobs of the actual buildbot worker that executes this script. Can be
+        overridden using the --jobs parameter so in case someone needs to
+        reproduce this build, they can adjust the number of jobs for the
+        reproducer platform. Alternatively, the worker can set the
+        BUILDBOT_JOBS environment variable or keep ninja/llvm-lit defaults.
+    always_clobber
+        Always clobber the build artifacts, i.e. disable incremental builds.
+    """
+
+    scriptpath = os.path.abspath(scriptpath)
+    llvmsrcroot = os.path.abspath(llvmsrcroot)
+    stem = pathlib.Path(scriptpath).stem
+    workdir_default = f"{stem}.workdir"
+
+    jobs_default = None
+    if jobs_env := os.environ.get("BUILDBOT_JOBS"):
+        jobs_default = int(jobs_env)
+    if not jobs_default:
+        jobs_default = workerjobs
+    if not jobs_default:
+        jobs_default = None
+
+    parser = parser or argparse.ArgumentParser(
+        allow_abbrev=True,
+        description="When executed without arguments, builds the worker's "
+        f"LLVM build configuration in {os.path.abspath(workdir_default)}. "
+        "Some build configuration parameters can be altered using the "
+        "following switches:",
+    )
+    parser.add_argument(
+        "--workdir",
+        default=workdir_default,
+        help="Use this dir as workdir to write the build artifact into. "
+        "--workdir=. uses the current directory.\nWarning: This directory "
+        "might be deleted",
+    )
+    parser.add_argument(
+        "--cachefile",
+        default=relative_if_possible(
+            pathlib.Path(scriptpath).with_suffix(".cmake"), llvmsrcroot
+        ),
```

Comment:
Is my understanding correct that by default this will pick up the `.cmake` cache file with the same name as the `.py` file for that worker? In this case: `polly-x86_64-linux-test-suite.cmake`

Reply:
That was possible here as well, it was just that the script could not change the default of `polly-x86_64-linux-test-suite.cmake`, so it had to ignore the user-provided cache file one if the default would be a different one.